Cpp_test/lib/ipc-pipe-receive.cpp

```diff
@@ -4,23 +4,31 @@ IPCPipeReceive::~IPCPipeReceive()
 {
 	close(this->pd);
 	unlink(this->info.method_name);
+	// Release semaphore
+	sem_post(this->sem);
+	sem_close(this->sem);
+	// Remove semaphores
+	sem_unlink(this->sem_name.c_str());
 }
 
 void IPCPipeReceive::init()
 {
 	this->file_handler.setup_file(info.file_name, FileMode::WRITE);
 	std::cout << "Waiting for sender... " << std::endl;
 	int time_wait{0};
+	std::string name;
```

Comment:
I still think this variable should not be called `name`. I did not test this, but could you not just add the slash to the semaphore name without using variable?